source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -0,0 +1,101 @@
+.. _fundamentals-records:
+
+=============================
+Document Data Format: Records
+=============================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to store and retrieve data in the {+driver-long+}
+using **Java records**. Java records are a type of Java class often used to
+model data and separate business logic from data representation.
+
+.. tip::
+
+   You can declare Java records in Java 14 or later. Learn more about the
+   functionality and restrictions of records from `Java 14 Language Updates: Records <https://docs.oracle.com/en/java/javase/14/language/records.html>`__.
+
+   If you are using an earlier version of Java, you can use plain old Java
+   objects instead. See the :ref:`<fundamentals-pojos>` guide for
+   implementation details.
+
+In this guide, you can learn how to perform the following tasks:
+
+- Write data from records into MongoDB.
+- Retrieve documents from MongoDB as records.
+
+.. _fundamentals-example-record:
+
+Serialize and Deserialize a Record
+----------------------------------
+
+You can provide the driver with instructions on how to serialize and
```

Comment:
It's better, but I still find this section confusing in the context of records.  We should either not mention codecs at all, or just say something like:

* The driver contains native support encoding and decoding Java records using the default codec registry (with a link to wherever we discuss that).